applications/dashboard/controllers/class.rolecontroller.php

```diff
@@ -133,6 +135,22 @@ public function edit($RoleID = false) {
         if ($this->Form->authenticatedPostBack() === false) {
             // Get the role data for the requested $RoleID and put it into the form.
             $Permissions = $PermissionModel->getPermissionsEdit($RoleID ? $RoleID : 0, $LimitToSuffix);
+
```

Comment:
Since we are adding this config setting as an optimization then it should be handled inside PermissionModel->getPermissionsEdit(), perhaps with an extra parameter. We don't want to process all of the categories if we don't have to or else the operation will never be able to be fully optimized.